src/main/java/org/jabref/gui/icon/IconTheme.java

```diff
@@ -2,64 +2,82 @@
+import org.kordamp.ikonli.materialdesign2.MaterialDesignK;
+import org.kordamp.ikonli.materialdesign2.MaterialDesignL;
+import org.kordamp.ikonli.materialdesign2.MaterialDesignM;
+import org.kordamp.ikonli.materialdesign2.MaterialDesignN;
+import org.kordamp.ikonli.materialdesign2.MaterialDesignO;
+import org.kordamp.ikonli.materialdesign2.MaterialDesignP;
+import org.kordamp.ikonli.materialdesign2.MaterialDesignR;
+import org.kordamp.ikonli.materialdesign2.MaterialDesignS;
+import org.kordamp.ikonli.materialdesign2.MaterialDesignT;
+import org.kordamp.ikonli.materialdesign2.MaterialDesignU;
+import org.kordamp.ikonli.materialdesign2.MaterialDesignV;
+import org.kordamp.ikonli.materialdesign2.MaterialDesignW;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import static java.util.EnumSet.allOf;
+
 public class IconTheme {
 
     public static final Color DEFAULT_DISABLED_COLOR = Color.web("#c8c8c8");
     public static final Color SELECTED_COLOR = Color.web("#50618F");
     private static final String DEFAULT_ICON_PATH = "/images/external/red.png";
     private static final Logger LOGGER = LoggerFactory.getLogger(IconTheme.class);
     private static final Map<String, String> KEY_TO_ICON = readIconThemeFile(IconTheme.class.getResource("/images/Icons.properties"), "/images/external/");
+    private static final Set<Ikon> ICON_NAMES = new HashSet<>();
 
-    public static void loadFonts() {
-        try (InputStream stream = getMaterialDesignIconsStream()) {
-            Font.loadFont(stream, 7);
-        } catch (IOException e) {
-            LOGGER.error("Error loading Material Design Icons TTF font", e);
-        }
-
-        try (InputStream stream = getJabRefMaterialDesignIconsStream()) {
-            Font.loadFont(stream, 7);
-        } catch (IOException e) {
-            LOGGER.error("Error loading custom font for custom JabRef icons", e);
-        }
+    public static Color getDefaultGroupColor() {
+        return Color.web("#8a8a8a");
     }
 
-    private static InputStream getMaterialDesignIconsStream() {
-        return IconTheme.class.getResourceAsStream("/fonts/materialdesignicons-webfont.ttf");
+    public static Optional<JabRefIcon> findIcon(String code, Color color) {
+        return ICON_NAMES.stream().filter(icon -> icon.toString().equals(code.toUpperCase(Locale.ENGLISH)))
+                         .map(internalMat -> new InternalMaterialDesignIcon(internalMat).withColor(color)).findFirst();
     }
 
-    private static InputStream getJabRefMaterialDesignIconsStream() throws IOException {
-        return IconTheme.class.getResource("/fonts/JabRefMaterialDesign.ttf").openStream();
+    public static Image getJabRefImageFX() {
+        return getImageFX("jabrefIcon48");
     }
 
-    public static Color getDefaultGroupColor() {
-        return Color.web("#8a8a8a");
-    }
+    public static void loadAllIkons() {
+        ServiceLoader<IkonProvider> providers = ServiceLoader.load(IkonProvider.class);
```

Comment:
Is this really the recommended way of loading an icon with Ikonli?

I had a quick look at the code and they seem to use some form of IkonResolver:
https://github.com/kordamp/ikonli/blob/efd8f31898e78434728e6d9e78b62f2474414dd3/core/ikonli-javafx/src/main/java/org/kordamp/ikonli/javafx/FontIcon.java#L235

Reply:
What's the difference between the description and the icon code?

Reply:
code is the unicode and description is the second parameter with the prefix e.g. mdi-

Reply:
ok, thanks. Then leave it like you have it right now.